// Copyright 2025 (c) By Rick Wesson & Support Intelligence, Inc.
// Licenced under the RLL 1.0
// protoc --go_out=.  --go-grpc_out=. holloman.proto
syntax = "proto3";
option go_package = "github.com/wessorh/HuntingHash";

package holloman ;

message ServiceCapabilities {
	string 		Acceleration = 10 ;
	int32		MaxOrder	 = 20 ;
	string		Magic		 = 30 ;
	bool		Ssdeep		 = 40 ;
} ;

message BufferRequest {
	bytes	Buffer = 10 ;
	string  Label  = 20 ;
} ;

message BufferResponse {
	int32	HOrder    	= 10 ;
	int32   Len			= 15 ;
	string  Id 			= 20 ;
	string  Magic     	= 30 ;
	string  Ssdeep      = 40 ;
	string  Sha1		= 50 ;
	string  Label       = 60 ;
	string  Tlsh		= 70 ;
	string  Sdhash		= 80 ;
} ; 

service Holloman {
	rpc Capabilities(ServiceCapabilities) returns(ServiceCapabilities) ;

	rpc ClusterBuffer(BufferRequest) returns(BufferResponse) ;
} ;

